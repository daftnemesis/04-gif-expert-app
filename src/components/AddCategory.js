import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

  //agregando estado para el input para la busqueda
  const [inputValue, setInputValue] = useState('')

  //handler para manejar el evento del input
  const handleInputChange = (e) => {
    setInputValue(e.target.value)
  }

  //handler para manejar el evento de enter y no refrescar la pagina
  //previniendo el comportamiento por defecto
  const handleSubmit = (e) => {
    e.preventDefault()

    if(inputValue.trim().length > 2) {
      setCategories(cats => [inputValue, ...cats])
      setInputValue('')
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <p> {inputValue}</p>
      <input 
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  )
}

//Agregando proptypes para hacer requerido el setCategories
AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}