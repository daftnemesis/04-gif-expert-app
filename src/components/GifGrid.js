import React from 'react'
import PropTypes from 'prop-types'

import { useFetchGifs } from '../hooks/useFetchGifs';
//import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

  //const [images, setImages] = useState([])

  // el hook useEffect se usa para hacer cambios en el estado pero solo cuando el componente es renderizado
  // useEffect( () => {
  //   getGifs(category)
  //     .then(img => setImages(img));
  // }, [category])

  const {data, loading} = useFetchGifs(category)


  return (
    <>
      <h3 className='animate__animated animate__fadeIn'>{category}</h3>

      {loading && <p className='animate__animated animate__flash'>Loading</p>}

      <div className='card-grid'>
          {
            data.map( img => (
              <GifGridItem 
                key={img.id}
                {...img}
              />
            ))
          }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}