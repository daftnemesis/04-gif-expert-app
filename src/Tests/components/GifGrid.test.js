import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid';

describe('Pruebas en GifGrid', () => {

  test('debe de coincidir el snapshot', () => {

    const wrapper = shallow(<GifGrid />);

    expect( wrapper ).toMatchSnapshot();
    
  });
  
});