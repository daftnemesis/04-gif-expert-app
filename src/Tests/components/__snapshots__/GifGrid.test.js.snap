// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid debe de coincidir el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  />
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
